### DISCUSSION TITLE
Welcome to Xomega.Net Discussions!

### DISCUSSION THREAD

**Discussion by XomegaNet** [Xomega-Net/Xomega.Net4VS#4] [Announcements]
## 👋 Welcome to the Xomega.Net community!

If you are new to Xomega.Net, then this is a place where you can feel free to:
- **ask questions** about our Xomega.Net platform;
- **browse questions and discussions** from other members to see if your question has already been answered;
- **share ideas for new features**, or **vote** on already suggested features;
- otherwise **engage** with both the **developers** and **other members** of our community.

💡 _We may use some questions or suggestions to create new issues and put them on our road map, in order to deliver a fix or enhancement in a future release._

If you are using Xomega.Net in your projects, then we encourage you to also:
- **help other members** by answering questions when possible;
- **help to set the direction** for the platform development by participating in polls and voting on suggestions.

To get started, comment below with an introduction of yourself and your project, and tell us how you think Xomega.Net platform might help you with your development.

**Comment by Liss199** [comment]
Hola, estoy haciendo uso de Xomega Solutions para un proyecto, al cargar la base y despues de generar las clases .xom dentro de la carpeta import, al generar el service layer los service implementations no se crea una clase y no se la razon por la que esto sucede me podrian ayudar

**Reply by XomegaNet** [reply]
Thanks for your interest in Xomega and for reporting this issue. We'd be happy to help you with this and any other issues you may encounter. I have also created a GitHub issue https://github.com/Xomega-Net/Xomega.Net4VS/issues/16 from your report, so that we could track it separately and provide additional comments there.

I assume that you are using the latest Xomega.Net version for Visual Studio 2022. In order to help us troubleshoot this issue, would you be able to send us to [our email](mailto:support@xomega.net) your solution with the projects, or at least the .Model project (.xomproj) that you use to generate the service implementations?

**Reply by XomegaNet** [reply]
Hi @Liss199, we haven't heard from you on this yet. Are you still experiencing this issue? If so, please get back to us with the details that we asked for, so that we could troubleshoot the problem on our end. Otherwise, we'll assume that you got this issue resolved and will go ahead and close the ticket https://github.com/Xomega-Net/Xomega.Net4VS/issues/16 that we opened for it.

**Comment by cagarraway** [comment]
Can you give any sort of roadmap as to if/when the following will be supported:

1. .NET 8
2. React/Vue/Svelte SPA
3. MAUI

**Reply by XomegaNet** [reply]
Hi @cagarraway, thanks for your interest in Xomega. Let me try to answer your questions as best as I can.

1. **.NET 8** - we are definitely planning to add .NET 8 support soon. While the current .NET 7 solution should be fully compatible with .NET 8, we are going to update the solution template and Xomega Framework to utilize .NET 8 and the new features that were introduced in there. We'll publish an announcement whenever full .NET 8 support is available, so you should get a notification if you registered on our website or follow us on X.

2. **React/Vue/Svelte SPA** - our current support for Typescript-based SPAs is limited to our XomegaJS framework, which is based on the now dated Knockout and Durandal frameworks. The problem with this architecture, as compared to Blazor WebAssembly, for example, is that you cannot reuse any of the C#-based UI code that is generated by Xomega or manually created using Xomega Framework, which allows you to write reusable UI code. So, in order to support JavaScript-based SPAs, we may need to build a separate JS framework first, which we are not planning to do in the nearest future. There are still ways to integrate Blazor components into other SPA applications though, so you can definitely leverage Xomega there that way. If there's a strong demand for more native support for such SPA architectures, we may consider implementing it in the future.

3. **MAUI** - currently the only way to use Xomega with MAUI is by using a hybrid mode and integrating Blazor views into the MAUI app. We can start by adding this architecture to our solution wizard, if you'd find it helpful. Xomega already supports generation of XAML views for WPF applications, so it should be possible to modernize it later to generate native MAUI views as well. We are going to explore these options, but it's not something that we have in our immediate plans. Again, depending on the interest from the community, we may prioritize this more.

I am not sure if you are just wondering about support for these technologies in general, or if you have specific interests in any of them in particular. If there is interest in specific new features or enhancements, perhaps it would be better to start a new discussion or issue for each one of them, so that people could discuss and vote on them separately.

Please don't hesitate to reach out to us with any questions or issues on our Xomega platform.

**Reply by cagarraway** [reply]
Well, firstly, thanks for replying so quickly. Part of the reason for my question was to to test your responsiveness, and you've impressed me with your reply's speed, depth and quality.

Whilst I don't have specific interests in the technologies mentioned, I was trying to discern how maintained this platform is, since I've not seen much comments in the discussion forum. I am very interested in using this platform for commercial development, but I needed to know how your plans to support future tech and some tech (like MAUI) I'm already using.

I've been considering building a platform like Xomega for quite a while, having used and built several code generators before. Specifically, I like the ability to model entities and operations in an agnostic manner, and add custom attributes to both, then generate various artifacts from them. However, it's clear that you've done a lot of good work already, so why reinvent the wheel?

I understand what you're saying about not being able to reuse c# UI code for SPA's, but I still think there is potential there. A SPA dev could still make use of the lower layers of application architecture generated, such as the data access, business code and REST API, assuming they wished to stick to the MS stack. Further have you considered creating a space where persons can upload custom generators designed to produce various special artifacts, or whole project types (such as a React client?) If you could get that going, it would not only re-engage your existing customers, but also draw interest to the whole platform, as the number of out-of-the-box items it can produce would grow and therefore be more attractive. This would end up benefitting everyone: you, because of increased revenue, and your customers, because of more surety of the continuity of the platform and business.

**Reply by XomegaNet** [reply]
Thanks for providing some context for your questions. Yes, we try to respond promptly to any inquiries, even on Thanksgiving. 😉 

We started Xomega using the now legacy WebForms, Silverlight, WPF and WCF technologies in VS2008. Yet, our technology-agnostic model and an architecture with clean layer separation allowed us to apply it to the new technologies, such as REST and Blazor, by just replacing the corresponding layer, which helped Xomega to stand the test of time. Not only does it show our commitment to continue evolving Xomega and keeping it up-to-date with modern developments, but it also demonstrates how Xomega can help you to future-proof your apps should the next great framework come along. You are absolutely right that modern JS-based SPAs can still very well leverage the business services developed using Xomega.

Xomega platform has been built with extensibility in mind, where you can add custom generators developed either by you or any third party. The full edition of Xomega.Net provides you with access to the source code of all generators, which would help you to customize the existing generators or build new ones. We would definitely love to improve our extensibility model, where developers could easily create and share "plugins" for generating certain artifacts. For the full experience, such a plugin may need to include a number of components - model enhancements, generators, documentation and possibly item or project templates, as well as any required NuGet packages. You're spot on about the benefits of such an open platform.

We would be happy to collaborate with you on making Xomega platform work for your apps and projects, including helping you to customize existing or build new generators, if needed.
